TauAnalysis/MCEmbeddingTools/python/customisers.py

```diff
@@ -0,0 +1,369 @@
+	#process.merge_step.remove(process.reducedEcalRecHitsEB)
+
+	process.merge_step.remove(process.ak4JetTracksAssociatorExplicit)
+
+	process.merge_step.remove(process.pfTrack)
+	process.merge_step.remove(process.pfConversions)
+	process.merge_step.remove(process.pfV0)
+	process.merge_step.remove(process.particleFlowDisplacedVertexCandidate)
+	process.merge_step.remove(process.particleFlowDisplacedVertex)
+	process.merge_step.remove(process.pfDisplacedTrackerVertex)
+	process.merge_step.remove(process.pfTrackElec)
+	process.merge_step.remove(process.electronsWithPresel)
+	process.merge_step.remove(process.mvaElectrons)
+	process.merge_step.remove(process.particleFlowBlock)
+	process.merge_step.remove(process.particleFlowEGamma)
+	process.merge_step.remove(process.gedGsfElectronCores)
+	#  process.merge_step.remove(process.gedGsfElectronsTmp)
+	process.merge_step.remove(process.gedPhotonCore)
+	process.merge_step.remove(process.ecalDrivenGsfElectronCores)
+	process.merge_step.remove(process.ecalDrivenGsfElectrons)
+	process.merge_step.remove(process.uncleanedOnlyElectronSeeds)
+	process.merge_step.remove(process.uncleanedOnlyAllConversions)
+	process.merge_step.remove(process.uncleanedOnlyPfTrack)
+	process.merge_step.remove(process.uncleanedOnlyPfTrackElec)
+	process.merge_step.remove(process.uncleanedOnlyGsfElectrons)
+	process.merge_step.remove(process.uncleanedOnlyElectronCkfTrackCandidates)
+	process.merge_step.remove(process.cosmicsVeto)
+	process.merge_step.remove(process.cosmicsVetoTrackCandidates)
+ #   process.merge_step.remove(process.ecalDrivenGsfElectronCores)
+ #   process.merge_step.remove(process.ecalDrivenGsfElectrons)
+ #   process.merge_step.remove(process.gedPhotonsTmp)
+ #   process.merge_step.remove(process.particleFlowTmp)
+	process.merge_step.remove(process.hcalnoise)
+
+	process.load('CommonTools.ParticleFlow.genForPF2PAT_cff')
+		
+	process.merge_step += process.genForPF2PATSequence
+	
+	process.schedule.insert(0,process.merge_step)
+	 # process.load('PhysicsTools.PatAlgos.slimming.slimmedGenJets_cfi')
+	
+	
+	return modify_outputModules(process, [keepMerged(dataTier)])
+
+def customiseMerging_Reselect(process, changeProcessname=True):
+	return customiseMerging(process, changeProcessname=changeProcessname, reselect=True)
+	
+################################ cross Customizers ###########################
+
+def customiseLHEandCleaning(process,reselect=False):
+	process._Process__name = "LHEembeddingCLEAN"
+	process = customiseCleaning(process,changeProcessname=False,reselect=reselect)
+	process = customiseLHE(process,changeProcessname=False,reselect=reselect)
+	return process
+
+def customiseLHEandCleaning_Reselect(process):
+	return customiseLHEandCleaning(process,reselect=True)
+
+################################ additionla Customizer ###########################
+
+def customisoptions(process):
+	try:
+		process.options.emptyRunLumiMode = cms.untracked.string('doNotHandleEmptyRunsAndLumis')
```

Comment:
this can be written as 
``` 
if not hasattr(process, "options") : process.options = cms.untracked.PSet()
process.options.emptyRunLumiMode = cms.untracked.string('doNotHandleEmptyRunsAndLumis')
```